src/app/utilities/AnnotateUtilities.ts

```diff
@@ -2,70 +2,101 @@
+            annotationFormFieldProperties[classProperty.label] = {
+                key: classProperty.key,
+                name: classProperty.label,
+                jsonPath: classProperty.value.replace(`$`, jsonPath),
+                type: classProperty.value.includes('has') ? 'array' : 'object',
+                properties: []
+            };
+
+            /* Add class values to form values */
+            const classValues: Dict | Dict[] = jp.value(superClass, classProperty.value.replace(`$`, jsonPath));
+            const classFormValues: Dict = { ...classValues };
+
+            if (!Array.isArray(classValues) && classValues) {
+                Object.entries(classValues).forEach(([key, value]) => {
+                    if (typeof (value) === 'object') {
+                        delete classFormValues[key];
+                    }
                 });
+
+                formValues[FormatFieldNameFromJsonPath(classProperty.value.replace(`$`, jsonPath))] = classFormValues;
+            } else {
+                formValues[FormatFieldNameFromJsonPath(classProperty.value.replace(`$`, jsonPath))] = classValues;
             }
 
-            parentNodes.forEach((parentNode, parentIndex) => {
-                parentClasses.push({
-                    jsonPath: jp.stringify(parentNode.path).replaceAll('"', "'"),
-                    name: MakeJsonPathReadableString(jp.stringify(parentNode.path)),
-                    ...(index > 0 && { parentName: MakeJsonPathReadableString(pathArray[index]) }),
-                    present: (Array.isArray(parentNode.value) && !!parentNode.value.length) || (!Array.isArray(parentNode.value) && typeof (parentNode.value) === 'object'),
-                    ...(parentNode.value.length && { options: parentNode.value.length }),
-                    ...(parentClasses[parentIndex]?.options && { dependent: true })
+            /* For each term of class, add it to the properties array of the corresponding class in the annotation form fields dictionary */
+            const termProperties = termsList.find(termsListOption => termsListOption.label === classProperty.label);
+
+            termProperties?.options.forEach(termOption => {
+                annotationFormFieldProperties[classProperty.label].properties?.push({
+                    key: termOption.key,
+                    name: termOption.label,
+                    jsonPath: termOption.value,
+                    type: 'string'
                 });
             });
-        }
+        });
     });
 
-    return parentClasses;
+    return {
+        annotationFormFieldProperties,
+        newFormValues: formValues
+    };
 };
 
+/**
+ * Function that returns all of the annotation motivations as options (excluding deletion as this is handled systematically)
+ * @param givenMotivation An already selected motivation that impacts the choice in motivation
+ */
+const GetAnnotationMotivations = (givenMotivation?: string) => ({
```

Comment:
can you make this a map/constant enum?

Reply:
?

Reply:
?

Reply:
Can you make this a map instead of recreating it every time in a function?

Reply:
You mean making it an import from a source file or something or declaring the variable at the top of the file so it does not get recreated every time?

Reply:
yeah something like that

Reply:
Well then I cannot do the check anymore for adding, let leave it like this